src/Language/Fixpoint/Types/Environments.hs

```diff
@@ -210,6 +210,9 @@ fromListIBindEnv = FB . S.fromList
 insertBindEnv :: Symbol -> SortedReft -> a -> BindEnv a -> (BindId, BindEnv a)
 insertBindEnv x r a (BE n m) = (n, BE (n + 1) (M.insert n (x, r, a) m))
 
+currentBindEnvId :: BindEnv a -> BindId
+currentBindEnvId (BE n _) = n
```

Comment:
Maybe
```
currentBindEnvSize :: BindEnv a -> Int
currentBindEnvSize (BE n _) = n
```
?

Reply:
Why? I would expect a BindEnv to be indexed by a BindId

Reply:
The commit that introduces the function says it returns the size of the BindEnv, and the implementation is returning the size of the BindEnv. In addition, it is unclear to me at least why the size needs to be framed as a BindId, and it is also unclear what bind such BindId would identify.

Reply:
It's returning the last BindId inserted I need it for inserting the local rewrite

Reply:
Thanks, I'm starting to understand. Here are a few problems I see with such way of doing it.

I don't think that's how inserting binds work. The size is alway 1 unit higher than the last inserted bind.
```Haskell
insertBindEnv :: Symbol -> SortedReft -> a -> BindEnv a -> (BindId, BindEnv a)                                                                                                                
insertBindEnv x r a (BE n m) = (n, BE (n + 1) (M.insert n (x, r, a) m))
```

Also, [the definition](https://github.com/ucsd-progsys/liquid-fixpoint/blob/b4d217a80c58814db1a607237f73a0edf0f0d706/src/Language/Fixpoint/Types/Environments.hs#L101) of `BindEnv` says that all the BindIds inside the BindEnv should be smaller than the size.

Also, I don't think such expectation could hold after using `deleteBindEnv`.

And lastly, I think I would get immediately the purpose of the function if it was called `idOflastInsertedBind`.

My guess is that a comment would need to make clear that the binding might no longer be present in the environment if it was deleted before calling `idOflastInsertedBind`, and that it should return `the size - 1`. But isn't a less surprising API to just have a function that returns the size of the environment and call it `bindEnvSize`?

Reply:
Ok fair tomorrow at the office I will implement your suggestion, I dont know about storing the index as the state is already pretty cluttered